elm/web/rhub.py

```diff
@@ -4,626 +4,788 @@
 import os
 import os.path
 import logging
-from urllib.request import urlopen
+import json
+import math
+import re
 import requests
 import pandas as pd
-from bs4 import BeautifulSoup
 
 logger = logging.getLogger(__name__)
 
 
-class ResearchOutputs():
-    """Class to handle publications portion of the NREL researcher hub."""
-    BASE_URL = "https://research-hub.nrel.gov/en/publications/?page=0"
-
-    def __init__(self, url, n_pages=1, txt_dir='./ew_txt'):
+class ProfilesRecord(dict):
+    """Class to handle a single profiles as dictionary data"""
+    def __init__(self, record):
         """
         Parameters
         ----------
-        url : str
-            Research hub publications URL, most likely
-            https://research-hub.nrel.gov/en/publications/
-        n_pages : int
-            Number of pages to get from the API. Typical response has 50
-            entries per page. Default of 1 ensures that this class doesnt hang
-            on a million responses.
-        txt_dir : str
-            File directory where you would like to save output .txt files.
+        record : dict
+            Profile in dict form, typically a response from the API.
         """
+        api_key = os.getenv("RHUB_API_KEY")
```

Comment:
In the class init docstrings for this class (and the other classes that use this) can you specify you need to set RHUB_API_KEY as an env variable and write a note on where to get this?

Reply:
I reached out to Rob about public access to the API and I'm still waiting to hear back so I just put in a placeholder for now.

Reply:
Can you just say "contact an NREL library representative" if this is how we get the key?